staging/src/k8s.io/apiserver/pkg/util/openstack/openstackutils.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2021 The Arktos Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"fmt"
+	"io/ioutil"
+	"k8s.io/klog"
+	"net/http"
+)
+
+const (
+	POD_JSON_STRING_TEMPLATE = `
```

Comment:
this is a good point. a simple template is ok however in this case, it would be better for code readability to use the Pod{} type to construct one here.

Reply:
replace or over write the object instead of replace the request body has other issues that require some deeper code change in the admission flow as well, which is something i wont try to change for now. put the template in a separated file and as input for this package to improve readability .